### ISSUE TITLE
prom_api.execute_maintenance - cannot update/delete rows from chunk "_hyper_29_146874_chunk" as it is compressed

### ISSUE THREAD

**Issue by DMOsipov** [timescale/promscale#454]
I've set up retention period for metric to 30 days.
When I try to execute maintenance, the ERROR occurres (see below).
I've copied the SQL query mentioned in the error message and tried to execute that SQL directly - and got the same error.
 
_hyper_29_146874_chunk - is the chunk of hypertable **prom_data**.node_network_receive_frame_total (prom_data, not prom_data_series)
I have no idea why TimescaleDB is trying to update prom_data table while executing this request.

```ERROR:  cannot update/delete rows from chunk "_hyper_29_146874_chunk" as it is compressed
CONTEXT:  SQL statement "
        WITH potentially_drop_series AS (
            SELECT distinct series_id
            FROM prom_data.node_network_receive_frame_total
            WHERE time < '2020-12-28 20:13:07.983624+03'
            EXCEPT
            SELECT distinct series_id
            FROM prom_data.node_network_receive_frame_total
            WHERE time >= '2020-12-28 20:13:07.983624+03' AND time < '2020-12-28 21:20:44.78165+03'
        ), confirmed_drop_series AS (
            SELECT series_id
            FROM potentially_drop_series
            WHERE NOT EXISTS (
                SELECT 1
                FROM  prom_data.node_network_receive_frame_total  data_exists
                WHERE data_exists.series_id = potentially_drop_series.series_id AND time >= '2020-12-28 21:20:44.78165+03'
                --use chunk append + more likely to find something starting at earliest time
                ORDER BY time ASC
                LIMIT 1
            )
        ) -- we want this next statement to be the last one in the txn since it could block series fetch (both of them update delete_epoch)
        UPDATE prom_data_series.node_network_receive_frame_total SET delete_epoch = current_epoch+1
        FROM _prom_catalog.ids_epoch
        WHERE delete_epoch IS NULL
            AND id IN (SELECT * FROM confirmed_drop_series)
    "
PL/pgSQL function mark_unused_series(text,timestamp with time zone,timestamp with time zone) line 6 at EXECUTE
SQL statement "SELECT _prom_catalog.mark_unused_series(metric_table, older_than, check_time)"
PL/pgSQL function drop_metric_chunks(text,timestamp with time zone,timestamp with time zone) line 48 at PERFORM
SQL statement "CALL _prom_catalog.drop_metric_chunks(r.metric_name, NOW() - _prom_catalog.get_metric_retention_period(r.metric_name))"
PL/pgSQL function execute_data_retention_policy() line 18 at CALL
SQL statement "CALL _prom_catalog.execute_data_retention_policy()"
PL/pgSQL function execute_maintenance() line 3 at CALL
SQL state: XX000```

**Comment by Harkishen-Singh** [comment]
Hey @DMOsipov, thank you for opening this issue. This is a bug on our side and is in our notice. We hope to release a fix for this soon.

**Comment by cevian** [comment]
Hi @DMOsipov, this looks a lot like a bug in timescaledb that was previously fixed in the following PR https://github.com/timescale/timescaledb/pull/2092 . That should have been included in timescaledb 1.7.3. What version of TimescaleDB are you running?

**Comment by DMOsipov** [comment]
Hi @cevian,

The error occured on TimescaleDB 1.7.2.
I upgraded to 1.7.4 after your comment, and the problem went away.
Thanks a lot for your kind support.
